src/extractor/guidance/intersection_handler.cpp

```diff
@@ -283,6 +285,47 @@ void IntersectionHandler::assignFork(const EdgeID via_edge,
     }
 }
 
+void IntersectionHandler::assignTrivialTurns(const EdgeID via_eid,
+                                             Intersection &intersection,
+                                             const std::size_t begin,
+                                             const std::size_t end) const
+{
+    for (std::size_t index = begin; index != end; ++index)
+        if (intersection[index].entry_allowed)
+            intersection[index].turn.instruction = {
+                findBasicTurnType(via_eid, intersection[index]),
+                getTurnDirection(intersection[index].turn.angle)};
+}
+
+void IntersectionHandler::assignCountingTurns(const EdgeID via_eid,
+                                              Intersection &intersection,
+                                              const std::size_t begin,
+                                              const std::size_t end,
+                                              const DirectionModifier modifier) const
+{
+    const constexpr TurnType turns[] = {TurnType::FirstTurn, TurnType::SecondTurn,
+                                        TurnType::ThirdTurn, TurnType::FourthTurn};
+    const constexpr TurnType ramps[] = {TurnType::FirstRamp, TurnType::SecondRamp,
+                                        TurnType::ThirdRamp, TurnType::FourthRamp};
+
+    const std::size_t length = end > begin ? end - begin : begin - end;
+    if (length > 4)
+    {
+        util::SimpleLogger().Write(logDEBUG) << "Counting Turn assignment called for " << length
+                                             << " turns. Supports at most four turns.";
+    }
```

Comment:
No. The function itself is only called from valid locations. It is mainly a debug assertion.